### PR TITLE
:hammer: Adding pool expansion and reboot scenarios

### PR DESCRIPTION
**Pull request by lsrinivas-pure**
**What this PR does / why we need it**:
Added tests for triggering pool expansion and reboting nodes while expansion is in progress

**Which issue(s) this PR fixes** (optional)
Closes # PTX-6893, PTX-6892

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/common.go [hunk 1]
```diff
@@ -3844,3 +3845,109 @@ func IsCloudDriveInitialised(n node.Node) (bool, error) {
...
+			if expandedPool.LastOperation.Status == opsapi.SdkStoragePool_OPERATION_FAILED {
+				return nil, false, fmt.Errorf("PoolResize has failed. Error: %s", expandedPool.LastOperation)
+			}
+		}
+		if expandedPool.LastOperation.Status == opsapi.SdkStoragePool_OPERATION_IN_PROGRESS {
+			// storage pool resize has been triggered
+			logrus.Infof("Pool %s expansion started", poolID)
+			return nil, true, nil
+		}
+		return nil, true, fmt.Errorf("pool %s resize not triggered ", poolID)
```
Comment (jainsmit) [inline tests/common.go:3867]: If this method invoked immediately after pool expand there could be few chances that pool will not be on above two status and it is simply online.  This method will not wait right?

Comment (lsrinivas-pure) [reply tests/common.go:3873]: yes, this logic is in rerty method, which checks for status to changes to in progress for a max of 2 mins in 10 second interval

File: tests/common.go [hunk 2]
```diff
@@ -3886,3 +3887,110 @@ func IsCloudDriveInitialised(n node.Node) (bool, error) {
...
+	err = Inst().V.WaitDriverUpOnNode(n, Inst().DriverStartTimeout)
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+//GetNodeWithGivenPoolID returns node having pool id
```
Comment (jainsmit) [inline tests/common.go:3959]: Is there any reason this method is not part of node driver?

File: tests/storage_pool/storage_pool_test.go [hunk 3]
```diff
@@ -143,7 +145,170 @@ var _ = Describe("{StoragePoolExpandDiskAdd}", func() {
...
+
+		var poolIDToResize string
+
+		pools, err := Inst().V.ListStoragePools(metav1.LabelSelector{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(len(pools)).NotTo(Equal(0))
+
+		// pick a random pool from a pools list and resize it
+		for _, pool := range pools {
```
Comment (jainsmit) [inline tests/storage_pool/storage_pool_test.go:296]: same

File: tests/common.go (hunk 2 above)
Comment (lsrinivas-pure) [reply tests/common.go:3959]: there is a dependency on GetStorageNodes function which inturn dependent on volume driver.
moving to node driver is giving dependency errors

File: tests/storage_pool/storage_pool_test.go [hunk 4]
```diff
@@ -143,7 +145,170 @@ var _ = Describe("{StoragePoolExpandDiskAdd}", func() {
...
```
[the remark this answers is not in the document]
Comment (lsrinivas-pure) [reply tests/storage_pool/storage_pool_test.go]: fixed

File: tests/storage_pool/storage_pool_test.go (hunk 3 above)
Comment (lsrinivas-pure) [reply tests/storage_pool/storage_pool_test.go]: fixed

**Review by jainsmit — APPROVED** [review]
LGTM !!

**Pull request merged**